File: common/src/main/java/net/yamac/springboot2/example/ExampleService.java
package net.yamac.springboot2.example;

import lombok.RequiredArgsConstructor;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.stereotype.Service;

@EnableConfigurationProperties(ExampleProperties.class)
@RequiredArgsConstructor
@Service
public class ExampleService {

    private final ExampleProperties exampleProperties;

    public String getMessage() {
        return exampleProperties.getMessage();
    }
}

File: common-kotlin/src/main/kotlin/net/yamac/springboot2/example/kotlin/ExampleService.kt
package net.yamac.springboot2.example.kotlin

import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.stereotype.Service

@EnableConfigurationProperties(ExampleProperties::class)
@Service
class ExampleService(private val exampleProperties: ExampleProperties) {

    fun getMessage(): String = exampleProperties.message
}
